### PR TITLE
Add ability to mark multiple notitications as read

### PR DESCRIPTION
**Pull request by cstns**
## Description

bulk mark read/unread notifications
- api endpoint
- ui

## Related Issue(s)

closes https://github.com/FlowFuse/flowfuse/issues/4511

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: forge/routes/api/userNotifications.js
Comment (Steve-Mcl) [inline forge/routes/api/userNotifications.js:40]: I realise this was an existing summary but this feels like it should be less specific

at minimum, it should say "as read or unread" since that functionality is already defined.

personally, i'd go with something like "Update notifications" or "Bulk update notifications" or similar

File: forge/db/models/Notification.js
Comment (Steve-Mcl) [inline forge/db/models/Notification.js:104]: why are you performing decode twice? (line 98 and line 104)

Also, not sure why the `Number.isNaN` is being used.  Perhaps you meant `ids = ids.map(id => typeof id === 'string' ? M.Notification.decodeHashid(id) : id)`

Lastly, as the array of data could be "bad ID strings", have you checked the consequences of the resulting `id` array on the datebase?

e.g. if `id` = `['goodId1', 'goodId2', 'badId1', 'badId2', 'goodId3']` what does the resulting array `id` hold and what affect does it have on the database?  This is something i would hope to see in a unit test.

**Review by Steve-Mcl — CHANGES_REQUESTED** [review]
couple of small issues Serban.

File: forge/routes/api/userNotifications.js
Comment (cstns) [reply forge/routes/api/userNotifications.js:40]: updated the summary entry with the recommended syntax. The route def was uncommented and did not remember to check it again

File: forge/db/models/Notification.js
Comment (Steve-Mcl) [inline forge/db/models/Notification.js:93]:
```suggestion
                updateNotificationState: async ({ read = true, ids = [] }, user) => {
```

Do we think, before this is committed, we should adjust this function name?

File: forge/routes/api/userNotifications.js
Comment (Steve-Mcl) [inline forge/routes/api/userNotifications.js:66]:
```suggestion
        await app.db.models.Notification.updateNotificationState(payload, request.session.User)
```

Only accept if above is accepted

**Review by Steve-Mcl — APPROVED** [review]
Approved but think we should rename the function as it implies only setting notifications as "read". 

You can chose.

**Comment by Steve-Mcl** [comment]
Also, since we are affecting APIs, we might want Nick @knolleary to glance over the schema before merging?

**Comment by Steve-Mcl** [comment]
PS, 

almost forgot:
* reference a changelog entry on the website repo in here somewhere (preferably the OP)
* are there any docs in this repo related to notification that need updating?

**Comment by cstns** [comment]
>     * reference a changelog entry on the website repo in here somewhere (preferably the OP)
>     * are there any docs in this repo related to notification that need updating?

Did not find any docs related to user notifications. I added a changelog pr which I referenced in a tasklist on the parent issue.

> Also, since we are affecting APIs, we might want Nick @knolleary to glance over the schema before merging?

Should I postpone merging the PR until we get a green light from @knolleary ?

**Comment by knolleary** [comment]
@cstns some quick feedback:
 - Please set a proper PR title as these end up in the changelog. For example, not having the issue number in the title and a clear short description of what this PR does - eg 'Add ability to mark multiple notitications as read'
 - When adding new external APIs please add a summary of the endpoint to the PR description to help review process - either in the PR or the related issue.

Please address the first item before merging - and take on the second item as feedback for future work.

**Comment by knolleary** [comment]
Depends what testing/verification its had between you and @Steve-Mcl - I think its relatively low risk to ship as-is.

**Comment by Steve-Mcl** [comment]
@cstns retested in pre-staging - still good to go (once merge and tests go green)

**Pull request merged**
